### PR TITLE
Adds improved headpatting

### PR DESCRIPTION
**Pull request by Supermichael777**
:cl:
add: Adds head-patting with tail recognition.
/:cl:

Oranges is a doof.  This is fundamentally superior to previous attempts and introduces more dynamics to character creation choices.

closes: #24062

### CODE REVIEW

**Comment by lzimann** [comment]
bad meme

**Comment by Thunder12345** [comment]
TEST **YOUR _FUCKING_**
# CODE

**Comment by Supermichael777** [comment]
this is a wholesome meme my friend

**Comment by ShadowDeath6** [comment]
Maybe if there're no PRs about patting then people will forget about it

**Comment by Iamgoofball** [comment]
# holy shit

# this has been denied 3 times knock it off

@optimumtact please close
@Cheridan please gitban

**Comment by Arianya** [comment]
>Posting the same PR 3 times is reason to be gitbanned

He who lives in glass houses, goof...

**Comment by Iamgoofball** [comment]
Stop using dot dot dot you fucking autist.

**Comment by Arianya** [comment]
Why...?

**Comment by PKPenguin321** [comment]
Because it indicates you're under 18

**Comment by Iamgoofball** [comment]
^^^^^^^^^^^^

**Comment by Arianya** [comment]
How terrible...

**Comment by Iamgoofball** [comment]
it's a bannable offence too

**Comment by Arianya** [comment]
I'll be very sad to get banned from goof's imaginary server where certain punctuation marks are bannable

**Comment by Arianya** [comment]
FUCK I FORGOT TO INCLUDE ONE

**Comment by Iamgoofball** [comment]
no, being under 18 is bannable you retarded fuck

**Comment by Arianya** [comment]
Thankfully my brain damage evidently saved me from the weird association between ellipses and age.

Keep on keepin on, goof.

**Comment by kevinz000** [comment]
How to get goofball to stop being a human being:
Add petting.

**Comment by ShadowDeath6** [comment]
>>goofball stop being a human being
>>>>>>>>>>>stop

**Comment by MMMiracles** [comment]
dot dot dot . . .

**Comment by Iamgoofball** [comment]
@ShadowDeath6 guess i gotta PR cattebeest again

**Comment by anconfuzedrock** [comment]
PAANNIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIC

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -210,6 +210,11 @@
 		if(lying)
 			M.visible_message("<span class='notice'>[M] shakes [src] trying to get [p_them()] up!</span>", \
 							"<span class='notice'>You shake [src] trying to get [p_them()] up!</span>")
+		else if(check_zone(M.zone_selected) == "head")
+			M.visible_message("<span class='notice'>[M] gives [src] a pat on the head to make [p_them()] feel better!</span>", \
+						"<span class='notice'>You give [src] a pat on the head to make [p_them()] feel better!</span>")
+			if(dna && dna.species && (("tail_lizard" in dna.species.mutant_bodyparts) || (dna.features["tail_human"] != "None")))
```
Comment (anconfuzedrock) [inline code/modules/mob/living/carbon/carbon_defense.dm]: honestly I am not smart but just to be sure shouldn't the "none" come after the first ) maybe I am looking at this wrong

Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/carbon_defense.dm]: what?

Comment (anconfuzedrock) [inline code/modules/mob/living/carbon/carbon_defense.dm]: I think I was confused forget it

**Comment by ExcessiveUseOfCobblestone** [comment]
Side note: I'm not sure if [this](https://github.com/Supermichael777/tgstation/blob/b3320a56f4fa2730d004295b530b2479cbb60dc5/code/modules/mob/living/carbon/carbon_defense.dm#L221) was intentional but why does hugging heal lol. Shaking up makes sense but patting someone's head or hugging them cures weaken/paralysis/etc.  Is that intended?

**Comment by ShadowDeath6** [comment]
The only time you'd be able to "heal" by hugging rather than shaking is if they were up and stunned by something like shadowling glare or the clock cult stun rune. 

Shaking someone should take priority over pattinng, though, for obvious reasons.

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 1 above)
Comment (Supermichael777) [inline code/modules/mob/living/carbon/carbon_defense.dm]: nope its checking that the slot dna.features that defines tail type is not type none.

**Comment by Supermichael777** [comment]
BEYOND THE SCOPE OF THIS PR; BALANCE CHANGE.

**Comment by Aliannera** [comment]
Even this is too weeb for me.

Then again I don't care about head pats. The hilarious drama around it sustains me however.

**Comment by Supermichael777** [comment]
hopeing for the jackpot

**Comment by lzimann** [comment]
<Cheridan> i appoint leoz as head of the head patting decision subcommittee

Denied.

**Comment by lzimann** [comment]
also kor merged the other one so ¯\_(ツ)_/¯

**Pull request closed without merging**